account_analytic_distribution_manual_date/tests/test_manual_distribution_date.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 (APSL - Nagarro) Bernat Obrador
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import date
+
+from freezegun import freeze_time
+
+from odoo.exceptions import UserError
+from odoo.tests import TransactionCase
+
+
+class TestManualDistributionDate(TransactionCase):
+    @classmethod
+    @freeze_time("2024-01-01")
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.manual_distribution = cls.env[
+            "account.analytic.distribution.manual"
+        ].create(
+            {
+                "name": "Test",
+                "start_date": "2024-01-01",
+                "end_date": "2024-12-31",
+            }
+        )
+
+        cls.invoice = cls.env["account.move"].create(
+            {
+                "move_type": "out_invoice",
+                "invoice_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Product A",
+                            "quantity": 1,
+                            "price_unit": 100.0,
+                            "manual_distribution_id": cls.manual_distribution.id,
+                        },
+                    )
+                ],
+            }
+        )
+
+    @freeze_time("2024-01-01")
+    def test_write_without_invoice_date(self):
+        """Test valid date within manual distribution range."""
+        self.invoice.write(
+            {
+                "invoice_line_ids": [
+                    (
+                        1,
+                        self.invoice.invoice_line_ids[0].id,
+                        {
+                            "manual_distribution_id": self.manual_distribution.id,
+                        },
+                    )
```

Comment:
The same here, and check the remaining code, please.

```suggestion
                    Command.update(
                        self.invoice.invoice_line_ids[0].id,
                        {
                            "manual_distribution_id": self.manual_distribution.id,
                        },
                    )
```